test/qa/lib/firewall.py

```diff
@@ -92,99 +95,231 @@
-            ip=server_ip, face=iface, subnet_addr=subnet
-        )
-
-    if protocol == "":
-        return """-A INPUT -i {face} -p udp -m udp --dport {p} -m comment --comment nordvpn -j ACCEPT
--A INPUT -i {face} -p udp -m udp --sport {p} -m comment --comment nordvpn -j ACCEPT
--A INPUT -i {face} -p tcp -m tcp --dport {p} -m comment --comment nordvpn -j ACCEPT
--A INPUT -i {face} -p tcp -m tcp --sport {p} -m comment --comment nordvpn -j ACCEPT
--A INPUT -i {face} -m connmark --mark 0xe1f1 -m comment --comment nordvpn -j ACCEPT
--A INPUT -i {face} -m comment --comment nordvpn -j DROP
--A OUTPUT -o {face} -p udp -m udp --dport {p} -m comment --comment nordvpn -j ACCEPT
--A OUTPUT -o {face} -p udp -m udp --sport {p} -m comment --comment nordvpn -j ACCEPT
--A OUTPUT -o {face} -p tcp -m tcp --dport {p} -m comment --comment nordvpn -j ACCEPT
--A OUTPUT -o {face} -p tcp -m tcp --sport {p} -m comment --comment nordvpn -j ACCEPT
--A OUTPUT -o {face} -m mark --mark 0xe1f1 -m comment --comment nordvpn -j CONNMARK --save-mark --nfmask 0xffffffff --ctmask 0xffffffff
--A OUTPUT -o {face} -m connmark --mark 0xe1f1 -m comment --comment nordvpn -j ACCEPT
--A OUTPUT -o {face} -m comment --comment nordvpn -j DROP""".format(
-            ip=server_ip, face=iface, p=port
-        )
-
-    return """-A INPUT -i {face} -p {proto} -m {proto} --dport {p} -m comment --comment nordvpn -j ACCEPT
--A INPUT -i {face} -p {proto} -m {proto} --sport {p} -m comment --comment nordvpn -j ACCEPT
--A INPUT -i {face} -m connmark --mark 0xe1f1 -m comment --comment nordvpn -j ACCEPT
--A INPUT -i {face} -m comment --comment nordvpn -j DROP
--A OUTPUT -o {face} -p {proto} -m {proto} --dport {p} -m comment --comment nordvpn -j ACCEPT
--A OUTPUT -o {face} -p {proto} -m {proto} --sport {p} -m comment --comment nordvpn -j ACCEPT
--A OUTPUT -o {face} -m mark --mark 0xe1f1 -m comment --comment nordvpn -j CONNMARK --save-mark --nfmask 0xffffffff --ctmask 0xffffffff
--A OUTPUT -o {face} -m connmark --mark 0xe1f1 -m comment --comment nordvpn -j ACCEPT
--A OUTPUT -o {face} -m comment --comment nordvpn -j DROP""".format(
-        ip=server_ip, face=iface, p=port, proto=protocol.lower()
-    )
-
-
-def is_active(port="", protocol="", subnet=""):
-    # Get interface name of your default gateway
-    output = sh.ip.route.show("default")
-    print(sh.ip.route.show())
-    _, _, _, _, iface = output.split(None, 5)
-
-    try:
-        # Get VPN server's IP address
-        status = sh.grep(sh.nordvpn.status(), "IP")
-        _, server_ip = status.split(None, 2)
-    except sh.ErrorReturnCode:
-        server_ip = ""
-    print("Default gateway:", iface, "Server's IP:  ", server_ip)
-
-    rules = _get_firewall_rules(
-        daemon.is_killswitch_on(), server_ip, iface, port, protocol, subnet
-    )
-    print("Expected rules:\n", rules)
+def _get_firewall_rules(ports: list[str]="", protocol: str="", subnets: list[str]="") -> list[str]:    
```

Comment:
I think it should rather get `None` as a default value, not an empty string. The same for `is_active()`.